instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/engine/MetricRegistrar.java

```diff
@@ -69,6 +73,12 @@ void enrollExtractor(
             ? metricInfo.getDescription()
             : attributeInfo.getDescription();
     String unit = metricInfo.getUnit();
+    String sourceUnit = metricInfo.getSourceUnit();
+
+    UnitConverter unitConverter = UnitConverterFactory.getConverter(sourceUnit, unit);
+    if (unitConverter != null) {
```

Comment:
[minor] it would be nicer to always return a no-op implementation rather than a potentially `null` value here, it removes another null-check in the callbacks too.